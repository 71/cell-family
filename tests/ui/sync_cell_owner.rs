cell_family::define! {
    type Family: CellOwner for Cell<T>;
}

fn main() {
    let mut owner = CellOwner::new();

    std::thread::spawn(|| {
        // We *cannot* reference the `owner` here (`CellOwner: !Sync`).
        let cell = Cell::new(vec![1, 2, 3]);

        cell.get_mut(&mut owner).push(4);
    });
}
