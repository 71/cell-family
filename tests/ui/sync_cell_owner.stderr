error[E0277]: the trait bound `Family: ThreadSafeFamily` is not satisfied
 --> tests/ui/sync_cell_owner.rs:8:5
  |
8 |     std::thread::spawn(|| {
  |     ^^^^^^^^^^^^^^^^^^ the trait `ThreadSafeFamily` is not implemented for `Family`
  |
  = note: required because of the requirements on the impl of `Send` for `cell_family::CellOwner<Family>`
  = note: 1 redundant requirement hidden
  = note: required because of the requirements on the impl of `Send` for `&mut cell_family::CellOwner<Family>`
  = note: required because it appears within the type `[closure@$DIR/tests/ui/sync_cell_owner.rs:8:24: 13:6]`
note: required by a bound in `spawn`
